orchestrator.py

```diff
@@ -0,0 +1,230 @@
+from __future__ import annotations
+import json
+import os
+import queue
+import time
+import uuid
+import multiprocessing as mp
+from dataclasses import dataclass, field, asdict, is_dataclass
+from typing import Any, Callable, Dict, List, Literal, Optional, Protocol, Set, Tuple
+from plan import PlanDiff, Plan, TaskInstance
+from task_registry import TaskRegistry, _REGISTRY
+from promise import SimpleViolation, MaxParallelism, scope_applies
+from event_system import EventWriter
+from execution_context import ExecutionContext
+
+class Orchestrator:
+    def __init__(self, registry: TaskRegistry, events_path: str = "outputs/events.jsonl"):
```

Comment:
```suggestion
    def __init__(self, registry: TaskRegistry, events_path: Path = Path("outputs/events.jsonl")):
```
also would prefer making the output a json list instead of json lines (but not important for now)